jca/WFLY-17180_Add_queryTimeout_validation_parameter.adoc

```diff
@@ -0,0 +1,72 @@
+= WFLY-17180 Add query-timeout validation parameter
+:author:            Tomasz Adamski
+:email:             tadamski@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          connector, validation
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.redhat.com/browse/
+
+== Overview
+
+Validation queries run without any explicit query timeout. If the validation query blocks (as it will with most drivers when a connection is invalid/stale), this can block the entire pool in certain cases (e.g. background-validation locks the pool during validation). Even with validate-on-match, the failure to enforce a short timeout can significantly impact time to get a connection if one or more failed connections are found in the pool.
+
+Setting a driver level query timeout is not a suitable workaround because this impacts normal queries that may be expected to run for significant periods. The timeout for normal queries needs to be different from what is tolerated for validation queries (which should respond nearly instantaneously or else the connection is likely broken).
+
+Minimally, an explicit query timeout should be supplied.
+
+It might also be useful if a pool property or perhaps a system property could support a configurable timeout period (e.g. to account for potential latency in some systems). The Apache Tomcat pool supports something of this nature with the validationQueryTimeout property.
+
+== Configuration
+
+'query-timeout' attribute would be added to validation tag of datasources subsystem:
+
+{code}
+   <datasource (...)>
+      (...)
+      <validation query-timeout='...'>
+          (...)
+      </validation>
+      (...)
+  </datasource>
+{code}
+
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}/WFLY-17180[WFLY-17180]
+
+=== Related Issues:
+
+* {issue-base-url}/EAP7-2166[EAP7-2166]
+
```

Comment:
Please add the new 'Stability' section as per, with the desired level selected.

https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc#stability-level

I suggest 'Community'.

If the selection is other than 'Default', please preface the L1 heading with "[the level"], e.g. "[Community]WFLY-17180 Add query-timeout validation parameter"